sympy/solvers/ode.py

```diff
@@ -2955,7 +2954,7 @@ def _constant_renumber(expr):
         newstartnumber maintains its values throughout recursive calls.
 
         """
-        global newstartnumber
+        nonlocal newstartnumber
```

Comment:
```
Compiling sympy/solvers/ode.py ...
  File "sympy/solvers/ode.py", line 2957
    nonlocal newstartnumber
                          ^
SyntaxError: invalid syntax
```

Reply:
from [here](https://stackoverflow.com/questions/14264313/syntax-error-on-nonlocal-statement-in-python): nonlocal only works in Python 3; it is a new addition to the language.

Reply:
Sorry, I really should have checked that before pushing. I've just pushed again